crates/red_knot_python_semantic/src/types.rs

```diff
@@ -301,7 +289,7 @@ impl<'db> Type<'db> {
     }
 
     #[must_use]
-    pub fn instance(&self) -> Type<'db> {
+    pub fn into_instance(&self) -> Type<'db> {
```

Comment:
What's the rationale for this name change?

Looking at https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv it seems to me that if we are adding a conventional prefix here, the appropriate one would be `to_`, as this is a conversion that stays at the same level of abstraction, and also doesn't consume `self` (`Type` is `Copy`).

Reply:
The name of the method has to change because it now conflicts with a generated `instance()` method that deriving `is_macro::Is` adds to the class: `ty.instance()` would produce `Some(<inner class type>)` if `ty` was a `Type::Instance` variant, or `None` if not.

> it seems to me that if we are adding a conventional prefix here, the appropriate one would be `to_`, as this is a conversion that stays at the same level of abstraction, and also doesn't consume `self` (`Type` is `Copy`).

Ahh, good point, `to_` is the better prefix here. I'll make that change, thanks!

Reply:
See https://docs.rs/is-macro/latest/is_macro/derive.Is.html#example. It's possible to rename the `is-macro`-generated methods, but here it feels easier to rename the custom method we already have